src/mapd-con-es6.js

```diff
@@ -1588,32 +1407,46 @@ class MapdCon {
     return curNonce
   }
 
+  renderVegaAsync = this.handleErrors(
+    (widgetid, vega, options) =>
+      new Promise((resolve, reject) => {
+        this.renderVega(widgetid, vega, options, (error, result) => {
+          if (error) {
+            reject(error)
+          } else {
+            resolve(result)
+          }
+        })
+      })
+  )
+
   /**
    * Used primarily for backend-rendered maps; fetches the row
    * for a specific table that was last rendered at a pixel.
    *
-   * @param {widgetId} Number The widget ID of the caller.
+   * @param {Number} widgetId The widget ID of the caller.
    * @param {TPixel} pixel The pixel. The lower-left corner is pixel (0,0).
-   * @param {String} tableName The table containing the geo data.
    * @param {Object} tableColNamesMap Map of the object of `tableName` to the array of column names.
-   * @param {Array<Function>} callbacks A collection of callbacks.
    * @param {Number} [pixelRadius=2] The radius around the primary pixel to search within.
+   * @param {Function} callback A callback function with the signature `(err, result) => result`.
+   *
+   * @returns {String} Current result nonce
    */
-
   getResultRowForPixel(
     widgetId,
     pixel,
     tableColNamesMap,
-    callbacks,
-    pixelRadius = 2
+    pixelRadius = 2,
+    callback = null
```

Comment:
Is there any danger of changing the argument order here? You probably already checked our other OmniSci code to verify, if this method is called, that the arguments match this new order. However, would this method ever be called by third-party code and, if so, will it break now that the argument order is different?

Reply:
Possibly. Third-party code will already break because of changing the signature of the Array `callbacks` param to a single flat function, to be sane and make the promisification simpler.

I'll note this change in the release notes, though, along with everything else (similar to the notes in the originating comment of this PR).

Reply:
that makes sense to me, though still makes me a bit nervous. I'm legitimately unsure how much external code might be affected here. Should we loop product or dev relations in? (I'm not super-familiar with how we interface with third parties here, so I'm unsure on the protocol.)